### PR TITLE
Avoid searching the pool for finding the best node, while we know the…

### PR DESCRIPTION
**Pull request by SaTahernejad**
… best node is node[0](just for AlpsSearchTypeBestFirst, AlpsSearchTypeBreadthFirst and AlpsSearchTypeHybrid)

### CODE REVIEW

File: Alps/src/AlpsSubTree.cpp [hunk 1]
```diff
@@ -1483,9 +1483,19 @@ AlpsSubTree::getBestNode() const
...
     AlpsTreeNode *bestNode = NULL;
 
-    bestNode = nodePool_->getBestNode();
-    bn2 = diveNodePool_->getBestNode();
-    
+    //Sahar:added:start
+    int searchStrategy = broker_->getModel()->AlpsPar()->entry(AlpsParams::searchStrategy);
+    const int msgLevel = broker_->getModel()->AlpsPar()->entry(AlpsParams::msgLevel);
+    if (msgLevel == 1000) {
+      std::cout << "correct version of Alps" << std::endl;
```
Comment (tkralphs) [inline Alps/src/AlpsSubTree.cpp]: What is the purpose of this here? I guess it doesn't belong.

**Comment by tkralphs** [comment]
Can you move this pull request to the main repo in the coin-or group?

File: Alps/src/AlpsSubTree.cpp (hunk 1 above)
Comment (SaTahernejad) [reply Alps/src/AlpsSubTree.cpp]: No. It was just like a comment for myself. I will modify it.

**Comment by SaTahernejad** [comment]
My current fork of Alps is from tkralphs/CHiPPS-ALPS. Because of conflicts, I cannot send pull request from my fork to coin-or/CHiPPS-ALPS. So, I will get a new fork from coin-or/CHiPPS-ALPS and send the pull request to this repository. Please correct me if I am wrong.
Update: I tried to get a fork from coin-or/CHiPPS-ALPS, but got the error that I have currently a fork of this repository. It seems that I should delete my current fork from  tkralphs/CHiPPS-ALPS, but I am afraid that doing so may cause issues for our bug report for Clp. Do you have any suggestions?

**Comment by tkralphs** [comment]
I'm not sure why there would be conflicts with the coin-or repo and not with mine. They should be identical. But in any case, I guess you should be able to pull the `master` branch from the coin-or repo into your existing fork, modify it, and then create a pull request from that. Just add the coin-or repo as a remote to your existing local repo. Does it make sense?

**Comment by SaTahernejad** [comment]
As I see the `master` branches of coin-or/CHiPPS-ALPS and tkralphs/CHiPPS-ALPS are not the same. the `master` branch of coin-or has 505 commits, while yours includes 453 commits. 
I think I understood what you mean. Let me check it.

**Comment by SaTahernejad** [comment]
As I am looking at the `master` branch of coin-or/CHiPPS-ALPS, it is very different from the `master` branch of tkralphs/CHiPPS-ALPS and also the `stable/1.5` branch (which is used by MibS). For example, in the `master` branch of coin-or, the `getBestNode()` function is defined in `AlpsNodePool.h`, but in the  `master` branch of tkralphs and also the `stable/1.5` branch, this function is defined in `AlpsNodePool.h`. However, the `stable/1.5` branches of both coin-or and tkralphs are the same. I think it is better that I send a pull request from my `stable/1.5` branch to this branch of coin-or. Do you agree with me?

**Comment by tkralphs** [comment]
No, the pull request should be against the master branch. I will take a look and see why they are so different. They shouldn't be.

**Comment by tkralphs** [comment]
OK, I had forgotten that the most recent stable version is `stable/2.0`, which corresponds to what is currently in `master`. I guess MibS is currently using `stable/1.5` because that is what `Blis` is built on. `stable/2.0` is not entirely backwards compatible so we can't easily upgrade. At some point in the future, we should switch to `DisCO` rather than `Blis` and then `DisCO` depends on `stable/2.0`. I'm assuming that the part of the code you're changing is the same in both `stable/1.5` and `stable/2.0` (as well as master) so it would be best to submit the PR against master and then I can cherry-pick it into `stable/1.5` and make a new release, as well as in `stable/2.0`. Make sense?

**Comment by tkralphs** [comment]
Now tkralphs/CHiPPS and coin-or/CHiPPS are identical, so it shouldn't matter which one you submit the request to.

**Comment by SaTahernejad** [comment]
Unfortunately, the parts of code that I should change in `stable/1.5` and `master`(or `stable/2.0`) are not the same. In `stable/1.5`, I should change the `getBestNode()` function in `AlpsNodePool.h`, while this function should be changed in `AlpsNodePool.cpp` in `master`.

**Comment by tkralphs** [comment]
OK, then you can submit a PR for `stable/1.5` and a separate one for `master` (if you don't mind). It looks like it doesn't actually matter what repo you forked. I can open a PR on the head fork against one of your branches, even though you did not originally clone the head fork (if you know what I mean). So if you just update your fork and make the changes there, you shouldn't need to do anything fancy like checking out a branch of the head fork into your current repo.

**Comment by SaTahernejad** [comment]
I modified both of `master` and `stable/1.5` branches in my fork. I can also directly send the PRs directly to the coin-or since there are no conflicts anymore. Do you suggest to send PRs to your fork or coin-or?

**Comment by tkralphs** [comment]
Of course to coin-or, so it will get automatically built and tested.

**Comment by SaTahernejad** [comment]
It's done.

**Pull request closed without merging**